### PR TITLE
New EasyConfig for OpenFOAM 3.0.0 with the foss 2016a toolchain

### PR DESCRIPTION
**Pull request by iotaka**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by iotaka** [comment]
Jenkins: ok to test

**Comment by boegel** [comment]
Jenkins: ok to test

@iotaka: I don't mind you talking to Jenkins, but he'll only listens to 'admins'  ;-)

**Comment by boegel** [comment]
lgtm

**Comment by boegel** [comment]
@iotaka: more missing easyconfigs:

```
'Irresolvable dependencies encountered:
METIS/5.1.0-foss-2016a-32bitIDX,
ncurses/5.9-foss-2016a,
SCOTCH/6.0.4-foss-2016a'
```

**Comment by iotaka** [comment]
ncurses/5.9-foss-2016a, come from old cmake used by metis

File: easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-3.0.0-foss-2016a.eb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    ('METIS', '5.1.0', '-32bitIDX'),
+    ('ncurses', '5.9'),
+    ('SCOTCH', '6.0.4'),
+]
+
+builddependencies = [
+    ('Bison', '3.0.4'),
+    ('CMake', '3.4.1'),
+    ('flex', '2.5.39'),
+    ('Boost', '1.60.0'),
```
Comment (wpoely86) [inline easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-3.0.0-foss-2016a.eb]: why is boost needed?

File: easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-3.0.0-foss-2016a.eb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    'OpenFOAM-%(version)s_cleanup.patch',
+    'OpenFOAM-%(version)s_libreadline.patch',
+    ('ThirdParty-%(version)s_cleanup.patch', ".."),  # patch should not be applied in OpenFOAM subdir
+]
+
+dependencies = [    
+    ('libreadline', '6.3'),
+    # OpenFOAM requires 64 bit METIS using 32 bit indexes (array indexes)
+    ('METIS', '5.1.0', '-32bitIDX'),
+    ('ncurses', '5.9'),
```
Comment (wpoely86) [inline easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-3.0.0-foss-2016a.eb]: only use 6.0 in 2016a toolchains

**Comment by iotaka** [comment]
Sorry @wpoely86 I forgot to update the ncurses version inside the main package!

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 9 out of 9 (6 easyconfigs in this PR)
Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/ee947d5bbd1ac3c6afbc for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 13 out of 13 (6 easyconfigs in this PR)
Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/8573ee18cc571f3c8649 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/47d96f3945255bad3391 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @iotaka

**Pull request merged**